### PR TITLE
Adding AppVeyor Support

### PR DESCRIPTION
**Pull request by garethellis0**
### Description
Adding AppVeyor support in preperation for a move to using our own CI hardware. This will not be required check for PR's for now, but we can run it in parallel in travis to help check that the same builds pass and fail on both.

*NOTE:* The AppVeyor config assumes #1184 is merged.

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Review by MathewMacDougall — DISMISSED** [review]
LGTM :+1: 

Do we have / need a team account for AppVeyor to manage anything?

**Comment by garethellis0** [comment]
> LGTM 
> 
> Do we have / need a team account for AppVeyor to manage anything?

Nope, it's a github app, you should be able to login and manage it.

**Review by jsn0716 — DISMISSED** [review]
LGTM 👍

**Review by jonathanlew — DISMISSED** [review]
Looks good. Wondering if we should do performance build on our own hardware

**Review by MathewMacDougall — DISMISSED** [review]

**Comment by garethellis0** [comment]
> Looks good. Wondering if we should do performance build on our own hardware

What does "performance build" mean here? Build the code with `-c opt`? Check how long compilation takes?

**Review by akhilveeraghanta — DISMISSED** [review]
LG:tm:

**Comment by garethellis0** [comment]
Holding off on this until #1184 is merged, just so we can make sure Appveyor properly passes.

**Review by EvanMorcom — DISMISSED** [review]
LGTM 💯

File: .appveyor.yml
Comment (MathewMacDougall) [inline .appveyor.yml]: nit: Can we not add random version numbers to the image name?

File: .appveyor.yml
Comment (MathewMacDougall) [inline .appveyor.yml:5]: non-blocking comment, but _maybe_ would be better if our machine was named more descriptively just so it's super clear what this is referring to for future people.

**Review by MathewMacDougall — CHANGES_REQUESTED** [review]

File: .appveyor.yml
Comment (garethellis0) [reply .appveyor.yml]: Removed, might need to go update the server tomorrow, we'll see if this builds.

File: .appveyor.yml
Comment (garethellis0) [reply .appveyor.yml:5]: At least added a comment about where the server is.

File: .appveyor.yml
Comment (garethellis0) [reply .appveyor.yml]: Ok, will need to go update the server tomorrow.

**Review by MathewMacDougall — APPROVED** [review]

**Review by akhilveeraghanta — APPROVED** [review]
LG:TM:

**Pull request merged**
